### ISSUE TITLE
Fcitx 5.1.6 crashes on uv_poll_start() on OpenBSD 7.4 current during dbus connection

### ISSUE THREAD

**Issue by ZhanYF** [fcitx/fcitx5#940]
**Describe the bug**
Fcitx 5.1.6 crashes instantly on OpenBSD 7.4 during dbus connection due to problem with uv_poll_start()

**To Reproduce**
1. Run `fcitx5-config-qt`
2. Click `Run fcitx 5`
3. `Segmentation fault (core dumped)`

Note that fcitx5 can be started, but if fcitx5-config-qt attempts to connect to it, it will crash.

**Expected behavior**
fcitx5 should start normally and allow the user to configure it.

**Desktop (please complete the following information):**
Seems not DE related, same behavior on KDE/Mate. 

**Additional context**
fcitx crash log:

```
I2024-01-17 05:44:53.244445 instance.cpp:1325] Override Enabled Addons: {}
I2024-01-17 05:44:53.245015 instance.cpp:1326] Override Disabled Addons: {}
I2024-01-17 05:44:53.250921 addonmanager.cpp:194] Loaded addon imselector
I2024-01-17 05:44:53.254690 addonmanager.cpp:194] Loaded addon quickphrase
I2024-01-17 05:44:53.263225 xcbconnection.cpp:189] Connecting to X11 display, display name::0.
I2024-01-17 05:44:53.264569 addonmanager.cpp:194] Loaded addon xcb
I2024-01-17 05:44:53.266212 addonmanager.cpp:194] Loaded addon clipboard
I2024-01-17 05:44:53.269322 addonmanager.cpp:194] Loaded addon pinyinhelper
I2024-01-17 05:44:53.270733 addonmanager.cpp:194] Loaded addon luaaddonloader
I2024-01-17 05:44:53.271696 addonmanager.cpp:194] Loaded addon unicode
I2024-01-17 05:44:53.289605 addonmanager.cpp:194] Loaded addon keyboard
I2024-01-17 05:44:53.293296 addonmanager.cpp:194] Loaded addon imeapi
I2024-01-17 05:44:53.304591 addonmanager.cpp:194] Loaded addon dbus
I2024-01-17 05:44:53.310039 addonmanager.cpp:194] Loaded addon ibusfrontend
I2024-01-17 05:44:53.321986 addonmanager.cpp:194] Loaded addon fcitx4frontend
I2024-01-17 05:44:53.325314 addonmanager.cpp:194] Loaded addon dbusfrontend
I2024-01-17 05:44:53.327816 addonmanager.cpp:194] Loaded addon xim
I2024-01-17 05:44:53.460389 inputmethodmanager.cpp:190] Found 617 input method(s) in addon keyboard
I2024-01-17 05:44:53.509526 addonmanager.cpp:194] Loaded addon kimpanel
I2024-01-17 05:44:53.510721 addonmanager.cpp:194] Loaded addon virtualkeyboard
I2024-01-17 05:44:53.652611 classicui.cpp:66] Created classicui for x11 display::0
I2024-01-17 05:44:53.652845 addonmanager.cpp:194] Loaded addon classicui
I2024-01-17 05:44:53.654360 addonmanager.cpp:194] Loaded addon notificationitem
I2024-01-17 05:44:53.656229 addonmanager.cpp:194] Loaded addon notifications

---- after launch fcitx5-config-qt ----

=========================
Fcitx 5.1.6 -- Get Signal No.: 11
Date: try "date -d @1705470293" if you are using GNU date ***
ProcessID: 32316
```

gdb trace:

```
(gdb) bt
#0  0x00000e67f8fc5623 in uv_poll_start () from /usr/local/lib/libuv.so.4.1
#1  0x00000e68e07616a1 in fcitx::LibUVSourceBase::resetEvent () from /usr/local/lib/libFcitx5Utils.so.1.0
#2  0x00000e68c442eba2 in check_write_watch () from /usr/local/lib/libdbus-1.so.11.3
#3  0x00000e68c442e090 in socket_do_iteration () from /usr/local/lib/libdbus-1.so.11.3
#4  0x00000e68c442cd1c in _dbus_transport_do_iteration () from /usr/local/lib/libdbus-1.so.11.3
#5  0x00000e68c4410919 in _dbus_connection_send_preallocated_unlocked_no_update () from /usr/local/lib/libdbus-1.so.11.3
#6  0x00000e68c440d363 in _dbus_connection_send_and_unlock () from /usr/local/lib/libdbus-1.so.11.3
#7  0x00000e68e07592d5 in fcitx::dbus::Message::send () from /usr/local/lib/libFcitx5Utils.so.1.0
#8  0x00000e6827c3d618 in _ZN5fcitx4dbus39ObjectVTablePropertyObjectMethodAdaptorINSt3__16vectorINS0_10DBusStructIJNS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEESA_SA_SA_SA_SA_bEEENS8_ISB_EEEENS2_5tupleIJEEENS_11Controller127availableInputMethodsMethodMUlDpOT_E_EEclENS0_7MessageE () from /usr/local/lib/fcitx5/libdbus.so
#9  0x00000e6827c3d4bc in _ZNSt3__110__function6__funcIN5fcitx4dbus39ObjectVTablePropertyObjectMethodAdaptorINS_6vectorINS3_10DBusStructIJNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEESC_SC_SC_SC_SC_bEEENSA_ISD_EEEENS_5tupleIJEEENS2_11Controller127availableInputMethodsMethodMUlDpOT_E_EEENSA_ISO_EEFbNS3_7MessageEEEclEOSQ_ () from /usr/local/lib/fcitx5/libdbus.so
#10 0x00000e68e074b7bb in fcitx::dbus::BusPrivate::objectVTableCallback () from /usr/local/lib/libFcitx5Utils.so.1.0
#11 0x00000e68e074dcc6 in fcitx::dbus::DBusObjectPathVTableMessageCallback () from /usr/local/lib/libFcitx5Utils.so.1.0
#12 0x00000e68c4424a04 in _dbus_object_tree_dispatch_and_unlock () from /usr/local/lib/libdbus-1.so.11.3
#13 0x00000e68c44104a7 in dbus_connection_dispatch () from /usr/local/lib/libdbus-1.so.11.3
#14 0x00000e68e074ed79 in std::__1::__function::__func<fcitx::dbus::Bus::attachEventLoop(fcitx::EventLoop*)::$_0, std::__1::allocator<fcitx::dbus::Bus::attachEventLoop(fcitx::EventLoop*)::$_0>, bool ()(fcitx::EventSource*)>::operator() () from /usr/local/lib/libFcitx5Utils.so.1.0
#15 0x00000e68e07611d6 in std::__1::__function::__func<fcitx::EventLoop::addDeferEvent(std::__1::function<bool ()(fcitx::EventSource*)>)::$_0, std::__1::allocator<fcitx::EventLoop::addDeferEvent(std::__1::function<bool ()(fcitx::EventSource*)>)::$_0>, bool ()(fcitx::EventSourceTime*, unsigned long long)>::operator() () from /usr/local/lib/libFcitx5Utils.so.1.0
#16 0x00000e68e0760626 in fcitx::TimeEventCallback () from /usr/local/lib/libFcitx5Utils.so.1.0
#17 0x00000e67f8fb783d in uv__run_timers () from /usr/local/lib/libuv.so.4.1
#18 0x00000e67f8fbcb95 in uv_run () from /usr/local/lib/libuv.so.4.1
#19 0x00000e68e0760121 in fcitx::EventLoop::exec () from /usr/local/lib/libFcitx5Utils.so.1.0
#20 0x00000e6855ef7d12 in fcitx::Instance::exec () from /usr/local/lib/libFcitx5Core.so.1.0
#21 0x00000e65e54f3a79 in main () from /usr/local/bin/fcitx5
```

[diagnose.output.txt](https://github.com/fcitx/fcitx5/files/13959294/diagnose.output.txt)
[fcitx5.core.zip](https://github.com/fcitx/fcitx5/files/13959326/fcitx5.core.zip)

Build options:
```
-DENABLE_WAYLAND=OFF 
-DCMAKE_INSTALL_SYSCONFDIR=/usr/local/share/examples
```

**Comment by wengxt** [comment]
should be fixed by https://github.com/fcitx/fcitx5/commit/12dc7bababc0f16a57561d397d21e0128b945e4c and included in 5.1.7 released today.

**Comment by ZhanYF** [comment]
Thanks for the swift response, could you also make a release on github?
